src/actions/package/providers/yay.rs

```diff
@@ -0,0 +1,95 @@
+use super::PackageProvider;
+use crate::actions::{package::PackageVariant, ActionError};
+use crate::utils::command::{run_command, Command};
+use serde::{Deserialize, Serialize};
+use std::process::{Command, Output, Stdio};
+use tracing::{debug, info, span, warn};
+use which::which;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Yay {}
+
+impl PackageProvider for Yay {
+    fn name(&self) -> &str {
+        "Yay"
+    }
+
+    fn available(&self) -> bool {
+        match which("yay") {
+            Ok(_) => true,
+            Err(_) => {
+                warn!(message = "yay not available");
+                false
+            }
+        }
+    }
+
+    fn bootstrap(&self) -> Result<(), crate::actions::ActionError> {
+        let span = span!(tracing::Level::INFO, "bootstrap").entered();
+
+        // Install base-devel and git to be able to pull and build/compile stuff
+        run_command(Command {
+            name: String::from("pacman"),
+            env: self.env(),
```

Comment:
`self.env()` isn't defined on this  `impl`.

You can use `HashMap::new()` instead, to keep it an empty environment